pkg/hs/urbit-king/lib/Urbit/Vere/Pier.hs

```diff
@@ -360,6 +361,28 @@ pier (serf, log) vSlog startedSig = do
   tExec <- acquireWorker "Effects" (router slog (readTQueue executeQ) drivz)
   tDisk <- acquireWorkerBound "Persist" (runPersist log persistQ execute)
 
+  -- Now that the Serf is configured, the IO drivers are hooked up, their
+  -- starting events have been dispatched, and the terminal is live, we can now
+  -- handle injecting events requested from the command line.
+  for_ (zip [1..] injected) $ \(num, ev) -> rio $ do
+    logTrace $ display @Text ("Injecting event " ++ (tshow num) ++ " of " ++
+                              (tshow $ length injected) ++ "...")
+    okaySig :: MVar (Either [Goof] ()) <- newEmptyMVar
+
+    let inject = atomically $ compute $ RRWork $ EvErr ev $ cb
+        cb :: WorkError -> IO ()
+        cb = \case
+          RunOkay _         -> putMVar okaySig (Right ())
+          RunSwap _ _ _ _ _ -> putMVar okaySig (Right ())
+          RunBail goofs     -> putMVar okaySig (Left goofs)
+
+    io inject
+
+    takeMVar okaySig >>= \case
+      Left goof -> error ("Goof in injected event: " <> show goof)
```

Comment:
What is the effect of `error` in this case? Does it kill the process?

Reply:
Yeah, it should unwind everything and kill the process. You can see another `error` earlier in this same function.

Reply:
This is a discrepancy with vere's behavior, which prints a stack trace but otherwise ignores errors here. I don't have strong feelings, and think we should let the hosting requirements drive this behavior, which presumably means killing the process. I just want to flag the discrepancy and confirm the requirements.

Otherwise, I have no issues with this approach (that aren't already captured in the TODO document re: parsing/validation).

Reply:
@brendanhay Do you have an opinion on the above, whether we should kill a king process when we fail an injected event?

Reply:
My preference would be an option/knob to choose the behaviour - otherwise I'd stick with vere's current behaviour.

For hosting the only current use-case is injecting non-critical events.

Reply:
Given that the current vere behaviour is to not die, I went and changed it so that it just logs the error instead of killing the process.